sb_arch_opt/algo/arch_sbo/models.py

```diff
@@ -303,7 +295,7 @@ def _get_design_variables(self) -> List['ds.DesignVariable']:
                     smt_des_vars.append(ds.FloatVariable(0, len(dv.options)-1))
                 else:
                     # Conditional categorical variables are currently not supported
-                    if is_dv_cond[i] and not self._global_disable_hierarchical_cat_fix:
```

Comment:
This was a temporary fix because SMT 2.1 didn't support categorical conditional (is_dv_cond) variables.
For SMT >= 2.2 --> `_global_disable... = True` --> `not _global... = False` --> the conditional variables are added as Categorical, not Integer variables.

So this if-statement can be replaced by just the line for adding the CategoricalVariable, because SMT >= 2.2 supports categorical conditional variables.